src/DocBlock/TypeExpression.php

```diff
@@ -393,7 +407,9 @@ private function parse(): void
             $consumedValueLength = \strlen($matches[0][0]);
             $index += $consumedValueLength;
 
-            if (\strlen($this->value) === $index) {
+            if (\strlen($this->value) <= $index) {
+                \assert(\strlen($this->value) === $index);
+
```

Comment:
Is this change actually needed? All tests pass without it. Tests aside, I don't fully get the logic behind it - entering condition for `<=` and then making an assert for `===`? If all `<` values are invalid there, why the `if` change?

Reply:
the logic is non trivial and critical - the `assert` is for safety only to assert the loop exists and always terminates